File: src/dataset/cpp/space-age_exercism-pawelo091991/gemini/GeminiFlash_space_age_v2.cpp
#include "space_age.h"

namespace space_age {

space_age::space_age(double seconds) : time(seconds) {}

double space_age::seconds() const {
  return time;
}

double space_age::on_earth() const {
  return time / earth_orbital_period;
}

double space_age::on_mercury() const {
  return time / (earth_orbital_period * mercury_orbital_period);
}

double space_age::on_venus() const {
  return time / (earth_orbital_period * venus_orbital_period);
}

double space_age::on_mars() const {
  return time / (earth_orbital_period * mars_orbital_period);
}

double space_age::on_jupiter() const {
  return time / (earth_orbital_period * jupiter_orbital_period);
}

double space_age::on_saturn() const {
  return time / (earth_orbital_period * saturn_orbital_period);
}

double space_age::on_uranus() const {
  return time / (earth_orbital_period * uranus_orbital_period);
}

double space_age::on_neptune() const {
  return time / (earth_orbital_period * neptune_orbital_period);
}

}  // namespace space_age

File: src/dataset/Java/space-age_exercism-uzilan/gemini/GeminiFlash_SpaceAge_v2.java
class SpaceAge {

    private final double seconds;
    private static final double EARTH_ORBITAL_PERIOD = 31557600.0;
    private static final double MERCURY_ORBITAL_PERIOD = 0.2408467 * EARTH_ORBITAL_PERIOD;
    private static final double VENUS_ORBITAL_PERIOD = 0.61519726 * EARTH_ORBITAL_PERIOD;
    private static final double MARS_ORBITAL_PERIOD = 1.8808158 * EARTH_ORBITAL_PERIOD;
    private static final double JUPITER_ORBITAL_PERIOD = 11.862615 * EARTH_ORBITAL_PERIOD;
    private static final double SATURN_ORBITAL_PERIOD = 29.447498 * EARTH_ORBITAL_PERIOD;
    private static final double URANUS_ORBITAL_PERIOD = 84.016846 * EARTH_ORBITAL_PERIOD;
    private static final double NEPTUNE_ORBITAL_PERIOD = 164.79132 * EARTH_ORBITAL_PERIOD;


    SpaceAge(double seconds) {
        this.seconds = seconds;
    }

    double onEarth() {
        return seconds / EARTH_ORBITAL_PERIOD;
    }

    double onMercury() {
        return seconds / MERCURY_ORBITAL_PERIOD;
    }

    double onVenus() {
        return seconds / VENUS_ORBITAL_PERIOD;
    }

    double onMars() {
        return seconds / MARS_ORBITAL_PERIOD;
    }

    double onJupiter() {
        return seconds / JUPITER_ORBITAL_PERIOD;
    }

    double onSaturn() {
        return seconds / SATURN_ORBITAL_PERIOD;
    }

    double onUranus() {
        return seconds / URANUS_ORBITAL_PERIOD;
    }

    double onNeptune() {
        return seconds / NEPTUNE_ORBITAL_PERIOD;
    }
}